src/backend/distributed/transaction/transaction_recovery.c

```diff
@@ -261,11 +262,27 @@ RecoverWorkerTransactions(WorkerNode *workerNode)
 			continue;
 		}
 
-		/* Check if the transaction is created by an outer transaction from a non-main database */
 		bool outerXidIsNull = false;
-		Datum outerXidDatum = heap_getattr(heapTuple,
-										   Anum_pg_dist_transaction_outerxid,
-										   tupleDescriptor, &outerXidIsNull);
+		Datum outerXidDatum = 0;
+		if (SearchSysCacheExistsAttName(DistTransactionRelationId(), "outer_xid"))
```

Comment:
I think this describes what we want better. It also doesn't call SearchSysCacheExistsAttName in the regular case when version checks are enabled.

```suggestion
		if (EnableVersionChecks || SearchSysCacheExistsAttName(DistTransactionRelationId(), "outer_xid"))
```